covid_model/fit_disconnected_co_regions.py

```diff
@@ -0,0 +1,92 @@
+import matplotlib.pyplot as plt
+import json
+import datetime as dt
+
+from db import db_engine
+from covid_model import RegionalCovidModel
+from model_fit import CovidModelFit
+from analysis.charts import actual_hosps, modeled
+from covid_model.cli_specs import ModelSpecsArgumentParser
+from data_imports import ExternalHosps
+
+all_regions = {
+    "cent": "Central",
+    "cm": "Central Mountains",
+    "met": "Metro",
+    "ms": "Metro South",
+    "ne": "Northeast",
+    "nw": "Northwest",
+    "slv": "San Luis Valley",
+    "sc": "South Central",
+    "sec": "Southeast Central",
+    "sw": "Southwest",
+    "wcp": "West Central Partnership"
+}
+
+
+def run():
+    # get fit params
+    parser = ModelSpecsArgumentParser()
+    parser.add_argument("-lb", "--look_back", type=int, help="the number of (default 14-day) windows to look back and refit; defaults to refitting all windows")
+    parser.add_argument("-bs", "--batch_size", type=int, help="the number of (default 14-day) windows to fit in each batch; default to running everything in one batch")
+    parser.add_argument("-is", "--increment_size", type=int, help="the number of windows to shift forward for each subsequent fit; default to 1")
+    parser.add_argument("-ws", "--window_size", type=int, help="the number of days in each TC-window; default to 14")
+    parser.add_argument("-tmn", "--tc_min", type=float, default=0, help="The lowest tc to allow")
+    parser.add_argument("-tmx", "--tc_max", type=float, default=0.99, help="The lowest tc to allow")
+    parser.add_argument("-rp", "--region_params", type=str, default="input/region_params.json", help="the path to the region-specific params file to use for fitting; default to 'input/region_params.json'")
+    parser.add_argument("-rg", "--regions", nargs="+", choices=all_regions.keys(), required=False, help="Specify the regions to be run, default is all regions (not counties)")
+    parser.add_argument("-wb", "--write_batch_output", action="store_true", default=False, help="write the output of each batch to the database")
+    parser.add_argument("-plt", "--plot", action="store_true", default=False, help="open a plot when fitting is complete, comparing modeled hosp to actual; default False")
```

Comment:
There's obviously a lot of repeated code here, with run_fit.py. I feel like it was a good call not to worry too much about that on the first pass, but we should maybe put this medium-high on the list for a refactor --- whether that means moving more of the shared content in CovidModelFit or some other kind of structure to reduce repetition.

Reply:
Good point, it was nice here to be able to parallelize the different runs, which might only be useful for regional model fitting and some scenario exploration stuff. Could we integrate that functionality into CovidModelFit?